staker/bold/bold_staker.go

```diff
@@ -322,10 +330,43 @@ func (b *BOLDStaker) Start(ctxIn context.Context) {
 				notifier.UpdateLatestConfirmed(confirmedMsgCount, *confirmedGlobalState)
 			}
 		}
+		err = b.updateStakerBalanceMetric(ctx)
+		if err != nil {
+			log.Warn("error updating staker balance metric", "err", err)
+		}
 		return b.config.AssertionPostingInterval
 	})
 }
 
+func (b *BOLDStaker) updateStakerBalanceMetric(ctx context.Context) error {
+	walletAddressOrZero := b.wallet.AddressOrZero()
+	if walletAddressOrZero != (common.Address{}) {
+		rollupUserLogic, err := boldrollup.NewRollupUserLogic(b.rollupAddress, b.client)
+		if err != nil {
+			return fmt.Errorf("error creating rollup user logic: %w", err)
+		}
+		amountStaked, err := rollupUserLogic.AmountStaked(&bind.CallOpts{Context: ctx}, walletAddressOrZero)
+		if err != nil {
+			return fmt.Errorf("error getting amount staked: %w", err)
+		}
+		boldStakerAmountStakedGauge.Update(arbmath.BigDivByUint(amountStaked, params.Ether).Int64())
+	} else {
+		boldStakerAmountStakedGauge.Update(0)
+	}
+
+	txSenderAddress := b.wallet.TxSenderAddress()
+	if txSenderAddress != nil {
+		balance, err := b.client.BalanceAt(ctx, *txSenderAddress, nil)
```

Comment:
If data already exists, is fine to expose as metrics, but I don't think it is worth adding additional queries just for metrics.

Reply:
We dont have this data already, since its not needed in the BoLD operations.

Though the request came from Jose https://offchainlabs.slack.com/archives/CTUJSH17W/p1747827876174549?thread_ts=1747218219.681789&cid=CTUJSH17W to have this metrics since we used to support the same in the old staker.

In the old staker as well we were making this call just to only fill data in metrics https://github.com/OffchainLabs/nitro/pull/1463/files#diff-61460c69c9f7736293e4ed7964854bfec78e190515e325197abc3537382cfcc5R760.

Also these call will only be made once every 15 minutes (AssertionPostingInterval).

Though I dont have a strong opinion here, happy to close the PR if these metrics are not that important.